pkg/mimir/mimir.go

```diff
@@ -663,16 +690,19 @@ func inheritFlags(log log.Logger, orig util.RegisteredFlags, dest util.Registere
 }
 
 type CommonConfig struct {
-	Storage bucket.StorageBackendConfig `yaml:"storage"`
+	Storage                bucket.StorageBackendConfig `yaml:"storage"`
+	ClusterValidationLabel string                      `yaml:"cluster_validation_label" category:"experimental"`
 }
 
 type CommonConfigInheritance struct {
-	Storage map[string]*bucket.StorageBackendConfig
+	Storage                map[string]*bucket.StorageBackendConfig
+	ClusterValidationLabel map[string]*grpcclient.Config
 }
 
 // RegisterFlags registers flag.
 func (c *CommonConfig) RegisterFlags(f *flag.FlagSet) {
 	c.Storage.RegisterFlagsWithPrefix("common.storage.", f)
+	f.StringVar(&c.ClusterValidationLabel, "common.cluster-validation-label", "", "Optionally define gRPC client's cluster validation label.")
```

Comment:
The fact that this one is `...cluster-validation-label` and the other one is `...cluster-validation.label` drives me a little bit crazy, tbh.